Traceback (most recent call last):
  File "/Users/mac/anaconda3/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/mac/anaconda3/lib/python3.11/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mac/anaconda3/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 173, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mac/anaconda3/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/mac/anaconda3/lib/python3.11/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/mac/anaconda3/lib/python3.11/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/mac/anaconda3/lib/python3.11/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
start_time = time(9, 31)
end_time = time(9, 45)
rank0=recordRank(basename,start_time,end_time)
rank0.head(20)
------------------

----- stdout -----
No record stored and not available from source.
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mUnboundLocalError[0m                         Traceback (most recent call last)
Cell [0;32mIn[15], line 3[0m
[1;32m      1[0m start_time [38;5;241m=[39m time([38;5;241m9[39m, [38;5;241m31[39m)
[1;32m      2[0m end_time [38;5;241m=[39m time([38;5;241m9[39m, [38;5;241m45[39m)
[0;32m----> 3[0m rank0[38;5;241m=[39mrecordRank(basename,start_time,end_time)
[1;32m      4[0m rank0[38;5;241m.[39mhead([38;5;241m20[39m)

Cell [0;32mIn[14], line 29[0m, in [0;36mrecordRank[0;34m(basename, start_time, end_time, sector)[0m
[1;32m     27[0m     rank[[38;5;124m'[39m[38;5;124m上涨家数[39m[38;5;124m%[39m[38;5;124m'[39m][38;5;241m=[39mrank[[38;5;124m'[39m[38;5;124m上涨家数[39m[38;5;124m'[39m][38;5;241m/[39m(rank[[38;5;124m'[39m[38;5;124m上涨家数[39m[38;5;124m'[39m][38;5;241m+[39mrank[[38;5;124m'[39m[38;5;124m下跌家数[39m[38;5;124m'[39m])
[1;32m     28[0m     columns[38;5;241m=[39m[[38;5;124m'[39m[38;5;124m排名[39m[38;5;124m'[39m,[38;5;124m'[39m[38;5;124m板块名称[39m[38;5;124m'[39m,[38;5;124m'[39m[38;5;124m板块代码[39m[38;5;124m'[39m,[38;5;124m'[39m[38;5;124m涨跌幅[39m[38;5;124m'[39m,[38;5;124m'[39m[38;5;124m市值比[39m[38;5;124m'[39m,[38;5;124m'[39m[38;5;124m换手率[39m[38;5;124m'[39m,[38;5;124m'[39m[38;5;124m上涨家数[39m[38;5;124m%[39m[38;5;124m'[39m,[38;5;124m'[39m[38;5;124m领涨股票[39m[38;5;124m'[39m,[38;5;124m'[39m[38;5;124m领涨股票-涨跌幅[39m[38;5;124m'[39m]
[0;32m---> 29[0m [38;5;28;01mreturn[39;00m rank[columns]

[0;31mUnboundLocalError[0m: cannot access local variable 'columns' where it is not associated with a value

